barcodes_generator_abstract/models/barcode_generate_mixin.py

```diff
@@ -36,15 +36,17 @@ class BarcodeGenerateMixin(models.AbstractModel):
     @api.model
     def create(self, vals):
         """It creates a new barcode if automation is active."""
-        barcode_rule = self.env['barcode.rule'].get_automatic_rule(self._name)
-        if barcode_rule.exists():
-            vals.update({
-                'barcode_rule_id': barcode_rule.id,
-            })
+        if not "barcode" in vals:
+            barcode_rule = self.env['barcode.rule'].get_automatic_rule(self._name)
+            if barcode_rule.exists():
+                vals.update({
+                    'barcode_rule_id': barcode_rule.id,
+                })
         record = super(BarcodeGenerateMixin, self).create(vals)
-        if barcode_rule:
-            record.generate_base()
-            record.generate_barcode()
+        if not "barcode" in vals:
+            if barcode_rule:
```

Comment:
I dont' understand why there is written one ``if barcode_rule`` and one ``if barcode_rule.exists()``
Do you know ? 

otherwise, (cosmetic point) I think that making a test with 2 conditions will make the diff smaller, and will be more readable.

     if barcode_rule:
         if not "barcode" in vals:

replaced by

    if barcode_rule and not "barcode" in vals:

(x times).
But just a matter of taste. not a blocking point.

Reply:
yes perfect i added some makeup